examples/simple/expected/Inputs/SomeInput.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Spawnia\Sailor\Simple\Inputs;
+
+class SomeInput
```

Comment:
what about having a constructor so I'm forced to provide a valid input?

Reply:
Given that input objects can be quite large and are generally unordered, I don't want to go with a constructor that takes positional arguments. I would be more inclined to do so if this library depended on PHP 8 with named arguments. Until then, I would rather leverage autocompletion and static validation and assign the properties individually like in the test code.

That said, I can see adding a method `public static fromArray(array $someInput): self`. There are some open questions about what to do with nested input objects in that case. Should we:
- expect them to be constructed by the user and simply pass them as-is?
- expect them to be passed in plain array form and instantiate them recursively?
- allow both previous options?

At that point, should we leverage an existing library such as https://github.com/spatie/data-transfer-object? Do you have alternative suggestions for a good base library or custom implementation?

Reply:
IMO fromArray would introduce too much complexity. 

I'd wait for named arguments on PHP 8. When I posted the comment I imagined exactly that named arguments are the solution to this.I'd not invest time to it now.

------------

I use https://github.com/spatie/data-transfer-object for DTOs that I use in gql resolvers and can recommend it.

Reply:
Agreed, postponed through https://github.com/spawnia/sailor/issues/40